GCC 4.0 is more picky about static declarations for variables previously
declared non-static.

From Andreas Jochens <aj@andaco.de>, for Debian bug #290365.

 dll/aim/toc/server.c |    4 ++--
 dll/aim/toc/toc.c    |    6 +++---
 include/modval.h     |   16 ++++++++--------
 source/ctcp.c        |    2 +-
 source/timer.c       |    2 +-
 5 files changed, 15 insertions(+), 15 deletions(-)

Index: ircii-pana-1.1/dll/aim/toc/server.c
===================================================================
--- ircii-pana-1.1.orig/dll/aim/toc/server.c	2003-04-10 21:09:06.000000000 -0400
+++ ircii-pana-1.1/dll/aim/toc/server.c	2005-07-27 22:34:21.000000000 -0400
@@ -7,14 +7,14 @@
 #include <unistd.h>
 #include "toc.h"
 
-static time_t lastsent = 0;
+time_t lastsent = 0;
 time_t login_time = 0;
 int my_evil;
 int is_idle = 0;
 int lag_ms = 0;
 int time_to_idle = 600;
 int is_away = 0;
-static struct timeval lag_tv;
+struct timeval lag_tv;
 
 void serv_add_buddy(char *name)
 {
Index: ircii-pana-1.1/dll/aim/toc/toc.c
===================================================================
--- ircii-pana-1.1.orig/dll/aim/toc/toc.c	2003-04-10 21:09:06.000000000 -0400
+++ ircii-pana-1.1/dll/aim/toc/toc.c	2005-07-27 22:34:21.000000000 -0400
@@ -39,9 +39,9 @@
 #include "toc.h"
 
 /* descriptor for talking to TOC */
-static int toc_fd;
-static int seqno;
-static unsigned int peer_ver=0;
+int toc_fd;
+int seqno;
+unsigned int peer_ver=0;
 int state;
 /* static int inpa=-1; */
 int permdeny = PERMIT_PERMITALL;
Index: ircii-pana-1.1/include/modval.h
===================================================================
--- ircii-pana-1.1.orig/include/modval.h	2003-04-10 21:09:07.000000000 -0400
+++ ircii-pana-1.1/include/modval.h	2005-07-27 22:34:21.000000000 -0400
@@ -650,7 +650,7 @@ extern Function_ptr *global;
 #define start_time ((time_t) *((time_t *)global[START_TIME]))
 #define idle_time() ((time_t) *((time_t *)global[IDLE_TIME]()))
 
-#define loading_global ((int) *((int *)global[LOADING_GLOBAL]))
+#define loading_global (*((int *)global[LOADING_GLOBAL]))
 #define target_window (*((Window **)global[TARGET_WINDOW]))
 #define current_window (*((Window **)global[CURRENT_WINDOW]))
 #define invisible_list (*((Window **)global[INVISIBLE_LIST]))
@@ -665,12 +665,12 @@ extern Function_ptr *global;
 #define dll_variable (*((IrcVariableDll **)global[DLL_VARIABLE]))
 #define dll_ctcp (*((CtcpEntryDll **)global[DLL_CTCP]))
 #define dll_window (*((WindowDll **)global[DLL_WINDOW]))
-#define window_display ((int) *((int *)global[WINDOW_DISPLAY]))
-#define status_update_flag ((int) *((int *)global[STATUS_UPDATE_FLAG]))
+#define window_display (*((int *)global[WINDOW_DISPLAY]))
+#define status_update_flag (*((int *)global[STATUS_UPDATE_FLAG]))
 #define tabkey_array (*((NickTab **)global[TABKEY_ARRAY]))
 #define autoreply_array (*((NickTab *)global[AUTOREPLY_ARRAY]))
-#define identd ((int) *((int *)global[IDENTD_SOCKET]))
-#define doing_notice ((int) *((int *)global[DOING_NOTICE]))
+#define identd (*((int *)global[IDENTD_SOCKET]))
+#define doing_notice (*((int *)global[DOING_NOTICE]))
 #define last_sent_msg_body (*((char **)global[LAST_SENT_MSG_BODY]))
 #define sent_nick (*((char **)global[SENT_NICK]))
 
@@ -698,9 +698,9 @@ extern Function_ptr *global;
 #ifdef GUI
 #ifndef MAIN_SOURCE
 #define lastclicklinedata ((char *) *global[LASTCLICKLINEDATA])
-#define contextx ((int) *((int *)global[CONTEXTX]))
-#define contexty ((int) *((int *)global[CONTEXTY]))
-#define guiipc ((int) *((int *)global[GUIIPC]))
+#define contextx (*((int *)global[CONTEXTX]))
+#define contexty (*((int *)global[CONTEXTY]))
+#define guiipc (*((int *)global[GUIIPC]))
 #endif
 #define gui_mutex_lock() ((void (*)(void)) global[GUI_MUTEX_LOCK])()
 #define gui_mutex_unlock() ((void (*)(void))global[GUI_MUTEX_UNLOCK])()
Index: ircii-pana-1.1/source/ctcp.c
===================================================================
--- ircii-pana-1.1.orig/source/ctcp.c	2003-05-27 03:00:22.000000000 -0400
+++ ircii-pana-1.1/source/ctcp.c	2005-07-27 22:34:21.000000000 -0400
@@ -176,7 +176,7 @@ CtcpEntryDll *dll_ctcp = NULL;
 
 /* CDE do ops and unban logging */
 
-static char	*ctcp_type[] =
+char	*ctcp_type[] =
 {
 	"PRIVMSG",
 	"NOTICE"
Index: ircii-pana-1.1/source/timer.c
===================================================================
--- ircii-pana-1.1.orig/source/timer.c	2003-04-10 21:09:07.000000000 -0400
+++ ircii-pana-1.1/source/timer.c	2005-07-27 22:34:21.000000000 -0400
@@ -148,7 +148,7 @@ BUILT_IN_COMMAND(timercmd)
  * This is put here on purpose -- we dont want any of the above functions
  * to have any knowledge of this struct.
  */
-static TimerList *PendingTimers;
+TimerList *PendingTimers;
 static char *schedule_timer (TimerList *ntimer);
 
 static char *current_exec_timer = empty_string;
